app/uk/gov/hmrc/bindingtariffclassification/scheduler/DaysElapsedJob.scala

```diff
@@ -40,8 +41,27 @@ class DaysElapsedJob @Inject()(appConfig: AppConfig, caseService: CaseService) e
   override def interval: FiniteDuration = FiniteDuration(jobConfig.intervalDays, TimeUnit.DAYS)
 
   override def execute(): Future[String] = {
-    caseService.incrementDaysElapsedIfAppropriate(jobConfig.intervalDays, appConfig.clock)
-      .map(count => s"Incremented the Days Elapsed for [$count] cases.")
+    ifTodayIsAWorkingDay { () =>
+      caseService.incrementDaysElapsed(jobConfig.intervalDays)
+        .map(count => s"Incremented the Days Elapsed for [$count] cases.")
+    }
+  }
+
+  private def ifTodayIsAWorkingDay(function: () => Future[String]): Future[String] = {
```

Comment:
I would return `Future[Boolean]`.
Then, where this method is called, I would do a `.map` and execute a block of code (your `function`) if it is `true`.

You can use `Logger.info` for the messages.
Also, it is a good idea to log in Kibana a message when the task started and a message when it finishes.
In the Platform Readiness Assessment it is a requirement for each scheduled job.

Reply:
I agree, will change